File: 백준/Silver/2108. 통계학/통계학.py
import sys

input = sys.stdin.readline
n = int(input())
nums = []
for _ in range(n):
  num = int(input())
  nums.append(num)

print(round(sum(nums) / len(nums)))
nums.sort()
print(nums[int(len(nums) / 2)])

numCount = {}
for num in nums:
  if not num in numCount:
    numCount[num] = 1
  else:
    numCount[num] += 1
numCountList = sorted(numCount.items(), key = lambda x : x[1], reverse = True)

frequency = numCountList[0][1]
mode = numCountList[0][0]
modeList = []

for num in numCountList:
  if num[1] == frequency:
    modeList.append(num[0])

modeList.sort()
if len(modeList) == 1:
  print(modeList[0])
else: 
  print(modeList[1])

print(max(nums) - min(nums))

File: 백준/Silver/2108. 통계학/통계학.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());

		int[] nums = new int[n];
		Map<Integer, Integer> map = new HashMap<>();
		
		double sum = 0;
		int min = Integer.MAX_VALUE;
		int max = Integer.MIN_VALUE;
		
		for (int i = 0; i < n; i++) {
			int num = Integer.parseInt(br.readLine());
			nums[i] = num;
			sum += num; 
			min = Math.min(min, num);
			max = Math.max(max, num);
			map.put(num, map.getOrDefault(num, 0) + 1);
		}
		
		Arrays.sort(nums);
		
		int maxCnt = 0;
		for (int num : map.values()) {
			maxCnt = Math.max(maxCnt, num);
		}
		
		List<Integer> maxCntList = new ArrayList<>();
		for (int key : map.keySet()) {
			if (map.get(key) == maxCnt) maxCntList.add(key);
		}
		Collections.sort(maxCntList);
		
		System.out.println(Math.round(sum / n));
		System.out.println(nums[n / 2]);
		
		if (maxCntList.size() >= 2) System.out.println(maxCntList.get(1));
		else System.out.println(maxCntList.get(0));
		
		System.out.println(max - min);
	}
}
